va/validation-authority.go

```diff
@@ -387,8 +387,9 @@ func (va *ValidationAuthorityImpl) validateTLSWithZName(identifier core.AcmeIden
 	}
 
 	challenge.Error = &probs.ProblemDetails{
-		Type:   probs.UnauthorizedProblem,
-		Detail: "Correct zName not found for TLS SNI challenge",
```

Comment:
This could use `probs.Unauthorized()` and we could add a `args ...interface{}` to it like `probs.Malformed()`.

Your call, though. We're still working out how to use this stuff.